src/compas/scene/scene.py

```diff
@@ -6,58 +6,95 @@
+        return [child.object for child in self.children]
+
+    def add_item(self, item, **kwargs):
+        sceneobject = SceneObject(item, **kwargs)
+        node = SceneObjectNode(sceneobject)
+        self.add(node)
+        sceneobject._node = node
+        return sceneobject
+
+
+class SceneTree(Tree):
+    def __init__(self, name=None):
+        super(SceneTree, self).__init__(name=name)
+        root = TreeNode(name="root")
+        self.add(root)
+
+    @property
+    def objects(self):
+        return [node.object for node in self.nodes if isinstance(node, SceneObjectNode)]
+
+    def add_object(self, sceneobject, parent=None):
+        node = SceneObjectNode(sceneobject)
+        if parent is None:
+            self.add(node, parent=self.root)
+        else:
+            parent_node = self.get_node_from_object(parent)
+            self.add(node, parent=parent_node)
+
+        sceneobject._node = node
+
+    def get_node_from_object(self, sceneobject):
+        for node in self.nodes:
+            if isinstance(node, SceneObjectNode):
+                if node.object == sceneobject:
+                    return node
+        raise ValueError("Scene object not in scene")
+
+
 class Scene(Data):
     def __init__(self, name=None, context=None):
         super(Scene, self).__init__(name)
-        self._tree = Tree("Scene")
+        self._tree = SceneTree("Scene")
         self.context = context
-        root = TreeNode(name="root")
-        self.tree.add(root)
 
     @property
     def tree(self):
         return self._tree
 
     @property
-    def sceneobjects(self):
-        return [node.attributes["sceneobject"] for node in self.tree.nodes if "sceneobject" in node.attributes]
+    def objects(self):
+        return self.tree.objects
 
     def add(self, item, parent=None, **kwargs):
         sceneobject = SceneObject(item, context=self.context, **kwargs)
-        name = item.name or item.__class__.__name__
-        node = TreeNode(name, attributes={"sceneobject": sceneobject})
-
-        if parent is None:
-            self.tree.add(node, parent=self.tree.root)
-        else:
-            parent_node = self._get_node(parent)
-            self.tree.add(node, parent=parent_node)
-
+        self.tree.add_object(sceneobject, parent=parent)
         return sceneobject
 
     def remove(self, sceneobject):
         node = self._get_node(sceneobject)
         self.tree.remove(node)
 
-    def _get_node(self, sceneobject):
-        for node in self.tree.nodes:
-            if "sceneobject" in node.attributes:
-                if node.attributes["sceneobject"] == sceneobject:
-                    return node
-        raise Exception("Scene object not in scene")
-
     def clear(self):
+        clear()
+
+    def clear_objects(self):
         guids = []
-        for sceneobject in self.sceneobjects:
+        for sceneobject in self.objects:
             guids += sceneobject.guids
             sceneobject._guids = None
         clear(guids=guids)
```

Comment:
Added a dedicated function `clear_objects` to clear objects managed by Scene, and  set `clear` to clear the entire context